test/404s.test.js

```diff
@@ -94,6 +99,18 @@ test('customized 404', t => {
         t.strictEqual(body.toString(), 'this was not found')
       })
     })
+
+    test('with error object', t => {
+      t.plan(3)
+      sget({
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/with-error'
+      }, (err, response, body) => {
+        t.error(err)
+        t.strictEqual(response.statusCode, 404)
+        t.strictEqual(body.toString(), 'this was not found')
```

Comment:
Why this request invokes the `setNotFoundHandler` ?

Reply:
because the error has a `statusCode` 404, this is _exactly_ the change this PR introduces.

Reply:
> (I need this for #724 to fully work)

Why?

If developer needs that the output is handled by the 404 route he should call `reply.notFound()`.

Reply:
https://github.com/fastify/fastify-static/blob/master/index.js#L34-L48 completely jumps over `reply.send()` and it avoids calling all of our hook system, just to call `reply.notFound()`. I think that's not acceptable, so here are #724 and #727.

Reply:
This PR can introduce some side effect.

Anyway I'm not against this PR